hms-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogServer.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class HMSCatalogServer {
+  private static final Logger LOG = LoggerFactory.getLogger(HMSCatalogServer.class);
+  private static Reference<Catalog> catalogRef;
+  static Catalog getLastCatalog() {
+    return catalogRef != null? catalogRef.get() :  null;
+  }
+
+  private HMSCatalogServer() {
+    // nothing
+  }
+
+  public static HttpServlet createServlet(SecureServletCaller security, Catalog catalog) throws IOException {
+    try (HMSCatalogAdapter adapter = new HMSCatalogAdapter(catalog)) {
+      return new HMSCatalogServlet(security, adapter);
+    }
+  }
+  
+  public static Catalog createCatalog(Configuration configuration) {
+    final String clazz = MetastoreConf.getVar(configuration, MetastoreConf.ConfVars.CATALOG_CLASS);
+    final Catalog catalog;
+    final String name;
+    String curi = configuration.get(MetastoreConf.ConfVars.THRIFT_URIS.getVarname());
+    String cwarehouse = configuration.get(MetastoreConf.ConfVars.WAREHOUSE.getVarname());
+    String cextwarehouse = configuration.get(MetastoreConf.ConfVars.WAREHOUSE_EXTERNAL.getVarname());
+    if ("HMSCatalog".equals(clazz)) {
+      name = "hms";
+      catalog = new HMSCatalog(configuration);
+    } else {
+      name = "hive";
+      catalog = new org.apache.iceberg.hive.HiveCatalog();
+      if (catalog instanceof Configurable) {
+        ((HiveCatalog) catalog).setConf(configuration);
+      }
+    }
+    Map<String, String> properties;
+    properties = new TreeMap<>();
+    if (curi != null) {
+      properties.put("uri", curi);
+    }
+    if (cwarehouse != null) {
+      properties.put("warehouse", cwarehouse);
+    }
+    if (cextwarehouse != null) {
+      properties.put("external-warehouse", cextwarehouse);
+    }
+    catalog.initialize(name, properties);
+    return catalog;
+  }
+
+  public static HttpServlet createServlet(Configuration configuration, Catalog catalog) throws IOException {
+    String auth = MetastoreConf.getVar(configuration, MetastoreConf.ConfVars.CATALOG_SERVLET_AUTH);
+    boolean jwt = "jwt".equalsIgnoreCase(auth);
+    SecureServletCaller security = new ServletSecurity(configuration, jwt);
+    Catalog actualCatalog = catalog;
+    if (actualCatalog == null) {
+      actualCatalog = createCatalog(configuration);
+      actualCatalog.initialize("hive", Collections.emptyMap());
+    }
+    catalogRef = new SoftReference<>(actualCatalog);
+    return createServlet(security, actualCatalog);
+  }
+
+  /**
+   * Convenience method to start a http server that only serves this servlet.
+   * @param conf the configuration
+   * @return the server instance
+   * @throws Exception if servlet initialization fails
+   */
+
+  public static Server startServer(Configuration conf, HMSCatalog catalog) throws Exception {
+    int port = MetastoreConf.getIntVar(conf, MetastoreConf.ConfVars.CATALOG_SERVLET_PORT);
+    if (port < 0) {
+      return null;
```

Comment:
It should be an exception imho.

Reply:
No, this is by design, if you don't specify anything in the configuration, the server does not start. If you put a 0, the system will find a free port; if you use > 0, it will attempt starting on this port.